### ISSUE TITLE
Backing up your Nodemcu to a file

### ISSUE THREAD

**Issue by rfordhamjr** [Aircoookie/WLED#479]
I have been using ESPHome-Flasher to flash my Nodemcu. Now that I have everything set the way I want it. Is there an app that will save it to a file to keep as a backup?

**Comment by sr1sws** [comment]
Read my mind. I came here to ask for a way, or if there is a way to export at least the presets. Would be nice to have sets of preset files for different holidays. A backup of the entire config would help too.

**Comment by Aircoookie** [comment]
In one of the next versions, you will be able to simply export `presets.json` and `config.json` files.
For now, you can read back the flash and save it to a bin file you could restore with either ESPHomeflasher or esptool. You will need to install [esptool](https://github.com/espressif/esptool), then run (replace PORT with the com port of your ESP, remove `-p PORT` if you don't know):
```
esptool.py -p PORT -b 460800 read_flash 0 0x400000 wled_backup.bin
```

Hope that helps :)

**Comment by sr1sws** [comment]
Great tip, thank you!

**Comment by Morphy99** [comment]
FYI [Tasmotizer](https://github.com/tasmota/tasmotizer/issues/39#issuecomment-674069258) *should* do this but it doesn't seem to be working for me on Windows.

**Comment by blazoncek** [comment]
Backup & Restore available since 0.13-b4 under Security & Updates settings.
Closing.
